File: app/src/main/java/com/example/starwars/model/Film.java
package com.example.starwars.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.google.gson.annotations.SerializedName;

import java.util.List;
@Entity
public class Film {
    @SerializedName("id")
    @PrimaryKey
    private int id;
    @SerializedName("title")
    private String title;
    @SerializedName("vote_average")
    private double voteAverage;
//    @SerializedName("genre_ids")
//    private int[] genres;
    @SerializedName("poster_path")
    private String posterPath;

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public double getVoteAverage() {
        return voteAverage;
    }

//    public int[] getGenres() {
//        return genres;
//    }

    public String getPosterPath() {
        return posterPath;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public void setVoteAverage(double voteAverage) {
        this.voteAverage = voteAverage;
    }

    public void setPosterPath(String posterPath) {
        this.posterPath = posterPath;
    }
}

File: app/src/main/java/com/example/starwars/model/Film.kt
package com.example.starwars.model

import androidx.room.Entity
import com.google.gson.annotations.SerializedName
import androidx.room.PrimaryKey
import com.example.starwars.model.Genre
import com.example.starwars.model.Crew
import io.reactivex.rxjava3.annotations.Nullable

@Entity
class Film {
    @SerializedName("id")
    @PrimaryKey
    var id = 0

    @SerializedName("title")
    var title: String? = null

    @SerializedName("vote_average")
    var voteAverage = 0.0

    @SerializedName("poster_path")
    var posterPath: String? = null

    @SerializedName("genres")
    var genres: @Nullable Array<Genre>? = null

    @SerializedName("overview")
    var overview: @Nullable String? = null

    @SerializedName("release_date")
    var releaseDate: @Nullable String? = null

    @SerializedName("production_companies")
    var crews: @Nullable MutableList<Crew>? = null
}